nav2_behavior_tree/plugins/action/back_up_action.cpp

```diff
@@ -29,9 +29,10 @@ class BackUpAction : public BtActionNode<nav2_msgs::action::BackUp>
 {
 public:
   BackUpAction(
+    const std::string & xml_tag_name,
     const std::string & action_name,
     const BT::NodeConfiguration & conf)
-  : BtActionNode<nav2_msgs::action::BackUp>(action_name, conf)
+  : BtActionNode<nav2_msgs::action::BackUp>(xml_tag_name, action_name, conf)
```

Comment:
```suggestion
  : BtActionNode<nav2_msgs::action::BackUp>(xml_tag_name, "back_up", conf)
```

If you just code the action name constant here like this, you can get rid of the "action_name" parameter, and the lambdas.

Reply:
Not if you want to support multiple instances of this with potentially other servers. This is now hardcoded into the codebase requiring a fork vs. changing the registration of the plugin.

Reply:
To clarify, this was done intentionally in the plugin registration so that if someone wants to change values, they just need to create a file with a new registration macro call with updated string. If its changed where you propose, it will be hardcoded for all users and they'd need to create their own new unique back up action codebase

Reply:
Doesn't #1365 allow them to change the action name in the XML file. This parameter just provides the default. In #1365, if someone sets a action name in the XML, it overrides this. No need to even create a new registration function

Reply:
Thats only for the follow path and computr path. This is for everything and on the registration level not the code level.

Reply:
Ok. I think I see what you're trying to do. I'm not convinced that anyone would ever want to do this, but alright.